### PR TITLE
17206 implement society NR flows

### PR DESCRIPTION
**Pull request by eve-git**
*Issue #:* /bcgov/entity#17206

*Description of changes:*
implement society NR flows.

new environment variable: **VUE_APP_SOCIETIES_ONLINE_HOME_URL**="https://dev.bcregistry.ca/societies/"
new feature flag: **enable-society**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: .env.example
Comment (severinbeauvais) [inline .env.example:12]: Also add this to devops/vaults.env so the CI action will populate it from 1Password.

Comment (severinbeauvais) [reply .env.example:12]: (This file is a sample file for developers to copy into `.env`, and is also the backup list of keys for PR temporary URLs if the cloudbuild script doesn't provide it.)

File: src/components/dialogs/pick-entity-or-conversion.vue
Comment (severinbeauvais) [inline src/components/dialogs/pick-entity-or-conversion.vue:97]: I think you can remove the Vue import now.

File: src/components/dialogs/pick-entity-or-conversion.vue
Comment (severinbeauvais) [inline src/components/dialogs/pick-entity-or-conversion.vue:233]: Please add a short comment to explain the business logic here (eg, why not support society?).

File: src/components/existing-request/nr-approved-gray-box.vue
Comment (severinbeauvais) [inline src/components/existing-request/nr-approved-gray-box.vue:28]: Could you rename the class?

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:574]: Please add a short comment to explain the business logic here.

Alternately, you could break up the logic into several, well-named variables.

File: src/plugins/featureFlags.ts
Comment (severinbeauvais) [inline src/plugins/featureFlags.ts:19]: This name confused me a bit until I went into LaunchDarkly and saw "namerequest support society NR in BC".

How about following the current convention and naming this `enable-society`?

**Review by severinbeauvais — COMMENTED** [review]
Looks very good and clean. Nice! Please see my small comments.

File: src/list-data/request-action-mapping.ts
Comment (eve-git) [reply src/list-data/request-action-mapping.ts:50]: Sorry I am not familiar with this part. It is old code.

File: src/components/existing-request/nr-approved-gray-box.vue
Comment (eve-git) [reply src/components/existing-request/nr-approved-gray-box.vue:28]: do you have suggestions?

Comment (severinbeauvais) [reply src/components/existing-request/nr-approved-gray-box.vue:28]: societies-online-button ?

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:574]: Should is `isSociety` ?

File: src/plugins/featureFlags.ts
Comment (severinbeauvais) [inline src/plugins/featureFlags.ts:19]: The default (or fallback) value should disable any new features, so that it's safe for prod users in case LaunchDarkly is not available (or in case they block LauchDarkly -- we don't want them to see new features by default).

File: src/list-data/request-action-mapping.ts
Comment (severinbeauvais) [reply src/list-data/request-action-mapping.ts:50]: Can you test around this code to see what it might do and whether it works correctly? Ask Mihai for help if needed.

**Review by severinbeauvais — APPROVED** [review]
LGTM but please get another review or 2.

File: src/mixins/common-mixin.ts
Comment (JazzarKarim) [inline src/mixins/common-mixin.ts:106]: Cool!

**Review by JazzarKarim — APPROVED** [review]
Looks great Eve! 👍

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-651-dk4f535f.web.app

**Comment by eve-git** [comment]
@severinbeauvais I've unit tested. @pwei1018 provided a temporary url for review. Does it mean the code merge will be applied after it has been tested?

**Comment by JazzarKarim** [comment]
> @severinbeauvais I've unit tested. @pwei1018 provided a temporary url for review. Does it mean the code merge will be applied after it has been tested?

Eve, the temporary URL is automatically generated after anyone commenting `/gcbru` (I didn't put an n in the end because it'll run the cloud build). The reason we do that so that other developers can check the work and in this case, it's a feature branch. The link will be used in UX Assurance and in QA.

**Comment by eve-git** [comment]
@JazzarKarim please commit /gcbrun again because I have a new commit. Thanks.
So, can I merge the code?

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-651-dk4f535f.web.app

**Comment by JazzarKarim** [comment]
> @JazzarKarim please commit /xxxxx again because I have a new commit. Thanks. So, can I merge the code?

Sorry Eve, I just saw this. Sure thing! Btw, you can also run the command yourself like just what happened 👍

**Review by ketaki-deodhar — COMMENTED** [review]
Looks good to me 👍

**Comment by severinbeauvais** [comment]
> @severinbeauvais I've unit tested. ... Does it mean the code merge will be applied after it has been tested?

Karim may have answered this... The PR temporary URL allows preliminary testing before the PR is merged. If you're asking about the feature branch -- we will need a PR to merge the feature branch into main, once everything is complete.

Ask again if you still have questions.

**Pull request merged**
